src/pages/payments/apply/Apply.tsx

```diff
@@ -177,105 +172,54 @@ export default function Apply() {
         </>
       )}
 
-      <Element leftSide={t('invoices')}>
+      <>
         {payment?.client_id ? (
-          <ComboboxAsync<Invoice>
-            endpoint={endpoint(
-              `/api/v1/invoices?payable=${payment?.client_id}&per_page=100`
-            )}
-            inputOptions={{
-              value: 'id',
-            }}
-            entryOptions={{
-              id: 'id',
-              value: 'id',
-              label: 'name',
-              searchable: 'number',
-              dropdownLabelFn: (invoice) =>
-                `${t('invoice_number_short')}${invoice.number} - ${t(
-                  'balance'
-                )} ${formatMoney(
-                  invoice.balance,
-                  payment.client?.country_id,
-                  payment.client?.settings.currency_id
-                )}`,
-            }}
-            onChange={({ resource }) =>
-              resource
-                ? handleInvoiceChange(
-                    resource.id,
-                    calcApplyAmount(resource.balance),
-                    resource.number
-                  )
-                : null
-            }
-            initiallyVisible={isLoading}
-            exclude={collect(formik.values.invoices)
-              .pluck('invoice_id')
-              .toArray()}
-            clearInputAfterSelection
-          />
+          <div className="flex flex-col px-4 sm:px-6 gap-y-2 mt-4">
+            <InputLabel>{t('invoices')}</InputLabel>
+
+            <DataTable<Invoice>
+              resource="invoice"
+              endpoint={`/api/v1/invoices?include=client&payable=${payment?.client_id}&per_page=100&sort=date|desc&per_page=1000`}
+              columns={columns}
+              onSelectedResourcesChange={(selectedResources) => {
+                if (selectedResources.length > 0) {
+                  const newInvoices: any[] = [];
+
+                  selectedResources.forEach((resource: Invoice) => {
+                    newInvoices.push({
+                      _id: v4(),
```

Comment:
I think it's safe for us to move to `window.crypto` in future for this since browser APIs can generate UUIDs now.

Reply:
@beganovich Replaced!